src/image.rs

```diff
@@ -190,4 +218,27 @@ mod test {
         let actual: TestImage = serde_yaml::from_str(yaml).unwrap();
         assert!(actual == expected);
     }
+
+    #[test]
+    fn test_image_spec_by_exec_spec() {
+        let yaml = r#"
+image:
+    exec:
+        command: foo
+        args:
+            - build
+        tag: "foobuild:1.0.0"
+"#;
```

Comment:
We should add a basic example to the docs - can just use `docker build` as the command.